src/shared/fdset.c

```diff
@@ -286,26 +330,27 @@ int fdset_close_others(FDSet *fds) {
 }
 
 unsigned fdset_size(FDSet *fds) {
-        return set_size(MAKE_SET(fds));
+        return hashmap_size(MAKE_HASHMAP(fds));
```

Comment:
what about this: we don't allow mixing indexed and non-indexed fds anyway, do we?

if we have fdset_is_indexed() we could halve the returned size if we are indexed.